.github/workflows/test.yml

```diff
@@ -2,12 +2,136 @@ name: test
 on:
   - push
   - pull_request
+
 jobs:
-  build:
+  install:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'npm'
+      - run: npm install
+
+  lint:
+    runs-on: ubuntu-latest
+    needs: install
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'npm'
+      - run: npm install
+      - run: npm run lint
+
+  typecheck:
+    runs-on: ubuntu-latest
+    needs: install
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'npm'
+      - run: npm install
+      - run: npx tsc
+
+  test-postgres:
+    runs-on: ubuntu-latest
+    needs: [install, lint, typecheck]
```

Comment:
I would not make these tests depend on the lint and typecheck jobs. I personally would still want to see the tests run even if the linter failed.